lib/id3c/api/routes.py

```diff
@@ -144,7 +144,30 @@ def receive_fhir(*, session):
 
     return "", 204
 
+@api_v1.route("/verify/barcode-uses", methods=['POST'])
+@content_types_accepted(["application/json"])
+@check_content_length
+@authenticated_datastore_session_required
+def verify_barcode_uses(*, session):
+    """
+    Validate a list of barcodes and use types.
+    
+    POST /barcode-uses with a JSON array of objects containing ``barcode`` and
+    ``use`` keys with string values.
+
+    Response will be a JSON array in the same order as input array, with objects containing
+    ``barcode`` (string) and ``use`` (string) from input array plus ``barcode_found`` (boolean)
+    to indicate if the barcode exists in the database, and ``use_match`` (boolean) to indicate
+    if the use from the input matches what is stored in the database (the ``use_match`` value
+    will be `null` if ``barcode_found`` is `false`).
+    """
+    barcode_use_list = request.get_json()
+
+    LOG.debug(f"Validating «{barcode_use_list}»")
 
+    result = datastore.verify_barcode_use_list(session, barcode_use_list)
+    return jsonify([ row._asdict() for row in result ])
```

Comment:
I know each row in the results has barcode_found and use_match. Do you think there would be benefit in inspecting the result and returning  an http response code accordingly here?

Reply:
Possibly, though I'm not sure what code we would want to apply since the result could always be a mix of verified and not verified. I can circle back to Jeremy on this so see what he prefers. 

I looked into possibility of implementing as a batch request, which could return a status code for each record, but that would require individual requests for each record combined into a single batch request, and most likely individual queries on the database for each record with results combined into a batch response. That seems much less efficient that this current method that uses a single select query.

Reply:
Fair point about batching. Sometimes a 207 is used for this (Multi-Status), indicating to look inside the response body for more detailed statuses about each operation or item. But, that's just one way of looking at it. I think the status could be interpretted in different ways. We could instead consider the response code indicative of the status of the verification operation, instead of pertaining to any individual item. We could keep a 200 response, as you have currently, if the verification operation completed successfully -- separating out the concern from whether barcodes were found/had matching uses. 

In that vein, it might be useful to the user to return a more specific 4XX type error, if the response is a malformatted array of json, or if doesn't contain the required keys, it would be ideal to return that to the user with a 400 response. I think currently it will just return a 5XX on any error.

Reply:
Ok, 207 may be a good option here, with individual status codes for each identifier/use pair from the submitted list (200:OK or 404:Not Found). 404 could indicate either the barcode doesn't exist, or the use type doesn't match, so there'd still be a need to check the response details to determine which it is.

As far as the request body goes, if we want to implement as a batch API the standard format would be more like this:
```
[
    {
        "method" : "GET",
        "path"   : "https://backoffice.seattleflu.org/production/api/v1/verification/barcode-uses/verify",
        "body"   : "{\"barcode\":\"abcd1234\",\"use\":\"sample\"}"
    },
    ...
]
```
That's much more verbose than what I have right now, and would lend itself more toward checking each barcode/use individually and merging the results, which requires a lot more queries as noted above. So I'm not sure it makes sense to go that route unless we really do need to return individual status codes.

As far as returning 4XX error, I don't think it's possible for this to return a malformated array of json, but I can update it to return a 4XX if the request body is malformatted. The SQL query simply converts the incoming array into rows and left joins the necessary tables for verifying the values, so the response will always include everything from the passed array, even maintaining the order and capitalization despite the matching being case insensitive.